cpg_workflows/stages/large_cohort.py

```diff
@@ -12,73 +14,110 @@
 class Combiner(CohortStage):
     def expected_outputs(self, cohort: Cohort) -> Path:
-        if vds_version := get_config()['workflow'].get('vds_version'):
-            vds_version = slugify(vds_version)
-            if not vds_version.startswith('v'):
-                vds_version = f'v{vds_version}'
-
-        vds_version = vds_version or get_workflow().output_version
-        return cohort.analysis_dataset.prefix() / 'vds' / f'{vds_version}.vds'
+        workflow_config = config_retrieve("workflow")
+        combiner_config = config_retrieve("combiner")
+        output_vds_name: str = slugify(
+            f"{workflow_config['cohort']}-{workflow_config['sequencing_type']}-{combiner_config['vds_version']}",
+        )
+        return cohort.analysis_dataset.prefix() / "vds" / f"{output_vds_name}.vds"
+
+    def get_vds_ids_output(self, vds_id: int) -> list[str]:
+        get_vds_analysis_query: DocumentNode = gql(
+            """
+            query getVDSByAnalysisIds($vds_id: Int!) {
+                analyses(id: {eq: $vds_id}) {
+                    output
+                    sequencingGroups {
+                        id
+                    }
+                }
+            }
+        """,
+        )
+        query_results: dict[str, Any] = query(get_vds_analysis_query, variables={"vds_id": vds_id})
+        vds_path: list[str] = [query_results["analyses"][0]["output"]]
+        vds_ids: list[str] = [sg["id"] for sg in query_results["analyses"][0]["sequencingGroups"]]
+        return vds_path + vds_ids
 
     def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
         # Can't import it before all configs are set:
         from cpg_workflows.large_cohort import combiner
 
-        j = get_batch().new_job('Combiner', (self.get_job_attrs() or {}) | {'tool': 'hail query'})
+        workflow_config = config_retrieve("workflow")
+        combiner_config = config_retrieve("combiner")
+        j: PythonJob = get_batch().new_python_job("Combiner", (self.get_job_attrs() or {}) | {"tool": "hail query"})
 
-        init_batch_args: dict[str, str | int] = {}
-        config = config_retrieve('workflow')
-
-        if config.get('highmem_workers'):
-            init_batch_args['worker_memory'] = 'highmem'
-        if config.get('highmem_drivers'):
-            init_batch_args['driver_memory'] = 'highmem'
-        if 'driver_cores' in config:
-            init_batch_args['driver_cores'] = config['driver_cores']
-        if not init_batch_args:
-            logging.warning(
-                "None of 'highmem_workers', 'highmem_drivers', or 'driver_cores' were specified in the config. If you're getting OOM errors, ensure these are included in the config.",
-            )
+        output_vds_path: Path = self.expected_outputs(cohort)
+        tmp_prefix = slugify(
+            f"{self.tmp_prefix}/{workflow_config['cohort']}-{workflow_config['sequencing_type']}-{combiner_config['vds_version']}",
```

Comment:
The config file allows for multiple cohorts to be used (`workflow.input_cohorts: list[str]`), and this is a `CohortStage`, so by definition it runs separately for each cohort. Once we get the ID/name shenanigans sorted you'd be better off using `cohort.name/id`, deriving it from the `Cohort` Target which is an input to this `queue_jobs` method, instead of going back to the config and picking up one config String.

That would hypothetically mean you can run this for several cohorts in parallel...

If you're not going to use it that way, this should be made into a MultiCohort stage (runs only once per analysis)

Reply:
Yes on the `cohort.name` (this is how it came up in the first place).

On Cohort vs MultiCohort, I'm not sure what direction PA wants. I'll ask @katiedelange and see.